### ISSUE TITLE
Add a linter or formatter

### ISSUE THREAD

**Issue by kasbah** [Hardmath123/nearley#310]
Add a linter and/or formatter. Something that requires zero configuration and just fixes style issues with a single invocation. 

Maybe one of: 
- [prettier](https://github.com/prettier/prettier)
- [standard](https://standardjs.com/)

Or even: 
- [prettier-standard](https://github.com/sheerun/prettier-standard)

Any preferences?

**Comment by tjvr** [comment]
I’m not generally a fan of linters; they tend to disagree with how I want my code to look :-)

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by kasbah** [comment]
In that case maybe you'd prefer to set up an eslint config that describes the current preferred style. The advantage of the ones I mentioned is that you don't have to spend the time doing that and you just drop them in and they do their thing. 

In general I would encourage you to get over the "how you want your code to look" as you always need to conform to the existing style when working on other people's code. I say: if it's consistent then be happy; anything else is a waste of time. 

Right now this code base, though it isn't too bad at all, is not consistent because there is no formatter in place.

**Comment by tjvr** [comment]
Yeah, it would be lovely if the codebase were more consistent :-)

I actually hadn't come across `prettier` before; I've just been experimenting with `prettier--no-semi --trailing-comma es5` for [Moo](https://github.com/tjvr/moo) and it _almost_ does what I want. I like that it operates on the AST, like `gofmt` (and unlike ESLint AFAIK).

However, this is @Hardmath123's project; and of the two of us, he tends to be more resistant to changes like this :-) Perhaps we could experiment with adding it to the tests, and perhaps we can come around to adding it to the core library later. How does that sound? Would you like to send a PR? (With the caveat that @Hardmath123 might reject it! :P)
